### ISSUE TITLE
[BUG] Add support for ezStringView properties

### ISSUE THREAD

**Issue by jankrassnigg** [ezEngine/ezEngine#819]
For properties we allow const char* getters and setters.
Internally somewhere, IIRC, we treat all const char* properties as ezString properties, so they do get storage.

We should also support ezStringView for passing around strings. Currently it isn't supported at all. For that you need to edit PropertyGridWidget.cpp.

```cpp
ezQtPropertyGridWidget::GetFactory().RegisterCreator(ezGetStaticRTTI<ezStringView>(), StandardTypeCreator);
ezQtPropertyGridWidget::GetFactory().UnregisterCreator(ezGetStaticRTTI<ezStringView>());

case ezVariant::Type::String:
case ezVariant::Type::StringView:
      return new ezQtPropertyEditorLineEditWidget();
```

Then it shows up in the UI, but the strings fail to be edited. I think it's missing the magic link that ezStringView should be treated like an ezString for storage.

**Comment by jankrassnigg** [comment]
After the ezVariant change to store ezStringViews internally, I've given this another try, see branch user/jk/string-view-property.
It partially works, but fails to restore the data correctly after a restart.

**Comment by TheophilusE** [comment]
Hey Jan, I also have this issue after using `ezStringView` extensively for reflection properties, while all of my unit tests passes, I discover that this issue is present when interaction with `ezStringView` properties from the editor standpoint. I can verify that this issue is not as a result of the underlying variant code, but as a consequence of what the string view represents. Since the variant implementation primarily assumes the `ezStringView` to be a view to a string that exists *somewhere* (which can be converted to an `ezString` in the `ezVariant` if needed), we see that the `ezQtPropertyEditorLineEditWidget` suffers from interpreting the broadcast event as a *fire-and-forget* mechanism, where it assumes that the ConvertedType when using `ezVariant.ConvertTo(m_OriginalType)` converts to its internal representation of the string data (which it does of course, if its not a an `ezStringView` with constructor ` bool bCopyString = false`).

You had mentioned that you thought there was some magic that the `const char*` should be treated as storage, but I have examined **every** reflection, variant, tools reflection, property widgets code and I did not find any special handling for `const char*`. The only *magic* is already in its implementation, where it constructs storage for the data contained using specialization.

From my findings, when we examine the code in `xiiQtPropertyEditorLineEditWidget::on_TextChanged_triggered` and `xiiQtPropertyEditorLineEditWidget::on_TextFinished_triggered`, the constructed variant is transient and thus causes the resulting `ezStringView` variant to be invalidated before it gets to `xiiQtPropertyEditorLineEditWidget::InternalSetValue`. One thing we could do is enable the following in `ConvertFunc` in Variant.cpp, but that defeats the purpose of having a lightweight view to a string somewhere being stored in an `ezVariant`.
```cpp
if constexpr (std::is_same_v<T, ezStringView>)
{
  m_Result = xiiVariant(result, true); // Change false to true, to enable copying of string view data for testing purposes.
}
else
{
  m_Result = result;
}
```

Another solution we could do is store an internal representation of the string in the `xiiQtPropertyEditorLineEditWidget`, which would allow just the construction of the variant to the view of the string, since its not runtime code and its event based, I doubt performance would be an issue there, as it is the same as doing it in the variant.
